lib/index.js

```diff
@@ -441,12 +427,34 @@ if (config.traceMode === 0) {
 //
 ao.traceMode = ao.cfg.traceMode;
 
-
+ao.notifications = undefined;
 //
 // the rest of the code is only relevant if bindings are loaded.
 //
 if (enabled) {
   const options = Object.assign({}, ao.cfg);
+  /* disable until requiring 'notifications' spikes memory is resolved.
+  const semverSatisfies = require('semver/functions/satisfies');
+
+  const badVersion = semverSatisfies(process.version, '>=12.0.0 <12.17.0');
+  //
+  // now start the notification service before initializing
+  // oboe (which might generate log messages).
+  //
+  if (options.unifiedLogging === 'preferred' && !badVersion || options.unifiedLogging === 'always') {
+    const Notifications = require('./notifications');
+    const notificationListener = require('./notifications-listener');
+    ao.notifications = new Notifications(notificationListener, bindings);
+    ao.notifications.startNotifier();
+  } else {
+    let msg = 'unified logging disabled by configuration';
+    if (options.unifiedLogging === 'preferred') {
+      msg += ` for node ${process.version}`;
```

Comment:
maybe i'm missing what you're saying but if `options.unifiedLogging` was `never` it is disabled by config. if it was preferred it will only be disabled when the node version has the RSS problem. that's what i'm trying to do anyway.